core/src/test/scala/unit/kafka/log/CleanerTest.scala

```diff
@@ -610,11 +611,88 @@ class CleanerTest extends JUnitSuite {
     assertEquals(-1, map.get(key(4)))
   }
 
+  @Test
+  def testCleanCorruptMessageSet() {
+    // This test verifies that messages corrupted by KAFKA-4298 are fixed by the cleaner
+
+    val log = makeLog()
```

Comment:
Do we need to set the compression codec to snappy for this log so that the cleaned data is compressed?